<launch>
  <arg name="robot" default="panda" doc="choose your robot. Possible values: [panda, fr3]"/>
  <arg name="arm_id" default="$(arg robot)" />
  
  <!-- Include the franka_control launch file with all arguments passed -->
  <include file="$(find franka_control)/launch/franka_control.launch" pass_all_args="true"/>

  <!-- Load parameters from the YAML file into the ROS parameter server -->
  <rosparam command="load" file="$(find franka_example_controllers)/config/franka_example_controllers.yaml" subst_value="true" />

  <!-- Start the controller spawner node to load and start 'my_controller' -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="my_controller"/>

  <!-- Start RViz with a specific configuration file -->
  <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find franka_example_controllers)/launch/rviz/franka_description_with_marker.rviz -f $(arg arm_id)_link0 --splash-screen $(find franka_visualization)/splash.png"/>

  <!-- Remove the interactive marker node -->
  <!-- <node name="interactive_marker" pkg="franka_example_controllers" type="interactive_marker.py" required="true" output="screen">
    <param name="link_name" value="$(arg arm_id)_link0" />
    <remap from="equilibrium_pose" to="/my_controller/equilibrium_pose" />
  </node> -->

  <!-- Start the pose publisher node -->
  <node name="pose_publisher" pkg="franka_example_controllers" type="pose_publisher.py" output="screen" />

  <!-- Start the rqt_reconfigure node for dynamic reconfiguration -->
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" required="false" />
</launch>
